src/main/resources/common/sitemap-entry.jsp

```diff
@@ -12,10 +12,12 @@
     <c:url var="localeUrl" value="${url.getBase(renderContext.site.language)}${urlNode.path}.html"/>
     <c:set var="finalUrl" value="${ (not empty vanityUrl) ? vanityUrl : localeUrl }"/>
     <%-- The URL host server name based on the input from sitemap UI panel--%>
-    <c:set var="urlHostServerName" value="${renderContext.site.getProperty('sitemapIndexURL').value}"/>
+    <c:set var="urlHostServerName" value="${renderContext.site.getPropertyAsString('sitemapIndexURL')}"/>
+    <c:set var="serverName" value="${sitemap:getServerName(urlHostServerName)}"/>
+
     <jcr:nodeProperty var="lastModified" node="${urlNode}" name="jcr:lastModified"/>
 
-    <loc>${urlHostServerName}${finalUrl}</loc>
+    <loc>${serverName}${finalUrl}</loc>
```

Comment:
I would try to default to url.server in case something goes wrong with parsing user input url

Reply:
Not sure if it is a good idea as it gives the illusion that there is a "valid" value when in fact the value input is invalid

Reply:
Either way I think we just need a way to inform users if anything is wrong with the URL. Logging is enabled for now in case anything goes wrong with parsing.